app/components/post_list/post/burn_on_read/unrevealed/unrevealed_burn_on_read_post.tsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+
+import {deletePost, revealBoRPost} from '@actions/remote/post';
+import Button from '@components/button';
+import {useServerUrl} from '@context/server';
+import {useTheme} from '@context/theme';
+import {PostModel} from '@database/models/server';
+import {getFullErrorMessage, isErrorWithStatusCode} from '@utils/errors';
+import {showBoRPostExpiredSnackbar} from '@utils/snack_bar';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    buttonBackgroundStyle: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.08),
+        height: 56,
+        marginBottom: 8,
+    },
+    buttonTextStyle: {
+        color: changeOpacity(theme.centerChannelColor, 0.56),
+    },
+}));
+
+type Props = {
+    post: PostModel;
+}
+
+export default function UnrevealedBurnOnReadPost({post}: Props) {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const handleRevealPost = useCallback(async () => {
+        const {error} = await revealBoRPost(serverUrl, post.id);
+        if (error && isErrorWithStatusCode(error) && error.status_code === 400) {
```

Comment:
what if the error is something else? a different status code or not even a server error but say there wasn't a connection at the time, if that is the case it seems there is no feedback for the growing in frustration user.

Reply:
Done. Always displaying the error message now and deleting when needed.